Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1094 @@
+                                                                                    @{N='ResourceName';E={$frontDoorEp.FrontDoorName}},
+                                                                                    @{N='DomainName';E={$frontDoorEp.DomainName}},
+                                                                                    @{N='CertificateType';E={$frontDoorEp.TypeOfCertificate}},
+                                                                                    @{N='MinimumTlsVersion';E={$frontDoorEp.MinTLSVersion}},
+                                                                                    @{N='isMinTLSVersionSetOnCustomDomain';E={$frontDoorEp.isMinTLSVersionSetOnCustomDomain}}
+                    }
+                    
+                    else
+                    {
+                        Write-Host "Skipping this Front End Point..." -ForegroundColor $([Constants]::MessageType.Warning)
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+                }
+              }
+            }
+            catch
+            {
+                $frontDoorsSkipped += $frontDoorResource
+                Write-Host "Error fetching Front Door configuration: Resource Name: [$($resourceName)], Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                $logResource.Add("ResourceName",($_.ResourceName))
+                $logResource.Add("Reason","Encountered error while fetching Front Door configuration")    
+                $logSkippedResources += $logResource
+                Write-Host "Skipping this resource..." -ForegroundColor $([Constants]::MessageType.Warning)
+            }
+        }
+    
+
+   foreach($item in $frontDoorsEndpointWithoutReqMinTLSVersion){
+        if (-not $frontDoorsEndpointWithoutReqMinTLSVersion)
+            {
+                Write-Host "Minimum TLS Version is set on the custom domains of front doors." -ForegroundColor $([Constants]::MessageType.Update)
+                Write-Host "Skipping this Front Door..." -ForegroundColor $([Constants]::MessageType.Warning)
+                Write-Host $([Constants]::SingleDashLine)
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                $logResource.Add("ResourceName",($_.ResourceName))
+                $logResource.Add("Reason","Minimum TLS Version is set on all the custom domains of front door.")    
+                $logSkippedResources += $logResource
+            }
+            else 
+            {
+                if (-not $frontDoorsEndpointWithoutReqMinTLSVersion.MinimumTlsVersion -ne $requiredMinTLSVersion)
```

Comment:
Is this a single object or collection $frontDoorsEndpointWithoutReqMinTLSVersion

Reply:
Collection of mutiple domains where TLS version is non-compliant.